inc/report.class.php

```diff
@@ -312,6 +312,237 @@ static function showDefaultReport() {
+    *
+    * @since 10.0.0
+   **/
+   static function newReportForNetworkInformations(
+      $from,
+      array $joincrit,
+      array $where = [],
+      array $select = [],
+      array $leftjoin = [],
+      array $innerjoin = [],
+      array $order = [],
+      $extra = ''
+   ) {
+      global $DB;
+
+      // This SQL request matches the NetworkPort, then its NetworkName and IPAddreses. It also
+      //      match opposite NetworkPort, then its NetworkName and IPAddresses.
+      // Results are groupes by NetworkPort. Then all IPs are concatenated by comma as separator.
+
+      if (count($joincrit) === 3) {
+         $andcrit = array_pop($joincrit);
+         $andcrit['AND']['PORT_1.is_deleted'] = 0;
+         $joincrit[] = $andcrit;
+      } else {
+         $joincrit[]['AND']['PORT_1.is_deleted'] = 0;
+      }
+
+      $criteria = [
+         'SELECT'       => array_merge([
+            'PORT_1.`itemtype` AS itemtype_1',
+            'PORT_1.`items_id` AS items_id_1',
+            'PORT_1.`id` AS id_1',
+            'PORT_1.`name` AS port_1',
+            'PORT_1.`mac` AS mac_1',
+            'PORT_1.`logical_number` AS logical_1',
+            new QueryExpression('GROUP_CONCAT(' . $DB->quoteName('ADDR_1.name') . ' SEPARATOR ' . $DB->quote(',') . ') AS ' . $DB->quoteName('ip_1')),
+            'PORT_2.`itemtype` AS itemtype_2',
+            'PORT_2.`items_id` AS items_id_2',
+            'PORT_2.`id` AS id_2',
+            'PORT_2.`name` AS port_2',
+            'PORT_2.`mac` AS mac_2',
+            new QueryExpression('GROUP_CONCAT(' . $DB->quoteName('ADDR_2.name') .' SEPARATOR ' . $DB->quote(',') . ') AS ' . $DB->quoteName('ip_2'))
+         ], $select),
+         'FROM'         => $from,
+         'INNER JOIN'   => $innerjoin + [
+            'glpi_networkports AS PORT_1' => [
+               'ON' => $joincrit
+            ]
+         ],
+         'LEFT JOIN'    => [
+            'glpi_networknames AS NAME_1' => [
+               'ON'  => [
+                  'PORT_1' => 'id',
+                  'NAME_1' => 'items_id', [
+                     'AND'    => [
+                        'NAME_1.itemtype'    => 'NetworkPort',
+                        'NAME_1.is_deleted'  => 0
+                     ]
+                  ]
+               ]
+            ],
+            'glpi_ipaddresses AS ADDR_1'  => [
+               'ON'  => [
+                  'NAME_1' => 'id',
+                  'ADDR_1' => 'items_id', [
+                     'AND'    => [
+                        'ADDR_1.itemtype'    => 'NetworkName',
+                        'ADDR_1.is_deleted'  => 0
+                     ]
+                  ]
+               ]
+            ],
+            'glpi_networkports_networkports AS LINK'  => [
+               'ON'  =>[
+                  'LINK'   => 'networkports_id_1',
+                  'PORT_1' => 'id', [
+                     'OR'     => [
+                        'LINK.networkports_id_2'   => $DB->quoteName('PORT_1.id')
```

Comment:
```suggestion
                        'LINK.networkports_id_2'   => new QueryExpression($DB->quoteName('PORT_1.id'))
```

Reply:
Not necessary in the condition part, this is handled